obspy/core/inventory/network.py

```diff
@@ -465,6 +468,19 @@ def select(self, station=None, location=None, channel=None, time=None,
                 if not fnmatch.fnmatch(sta.code.upper(),
                                        station.upper()):
                     continue
+
+            if public_id is not None:
+                if hasattr(sta, 'extra'):
+                    if hasattr(sta.extra, 'publicID'):
+                        if not fnmatch.fnmatch(
+                            sta.extra['publicID'].value.upper(),
+                            public_id.upper()
```

Comment:
I'm not sure if `publicID` in sc3ml is case sensitive, but personally, I'd probably leave the `upper()` away. I know it's done like this in some other comparisons, and I'm not sure if I agree with that but it is what it is and thats more coming from the SEED perspective where everything was supposed to be all uppercase characters only anyway.